### ISSUE TITLE
`VarChecker` produces `null` fix

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#567]
Consider this class:
```java
import com.google.common.io.ByteSource;
import java.io.IOException;
import java.io.InputStream;

final class Dummy {
    public void process(ByteSource byteSource) throws IOException {
        // Error is also triggered if `is` is redundantly declared `final`.
        try (InputStream is = byteSource.openStream()) {
            /* Do something. */
        }
    }
}
```

When running Error Prone with `-XepPatchChecks:Var -XepPatchLocation:IN_PLACE`, the following exception occurs:
```
/path/to/Dummy.java:8: error: An unhandled exception was thrown by the Error Prone static analysis plugin.
        try (InputStream is = byteSource.openStream()) {
                         ^
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: 2.0.18
     Stack Trace:
     java.lang.IllegalArgumentException: fix must not be null
  	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
  	at com.google.errorprone.matchers.Description$Builder.addFix(Description.java:192)
  	at com.google.errorprone.bugpatterns.VarChecker.handleLocalOrParam(VarChecker.java:94)
  	at com.google.errorprone.bugpatterns.VarChecker.matchVariable(VarChecker.java:72)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitVariable(ErrorProneScanner.java:1195)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitVariable(ErrorProneScanner.java:144)
  	at com.sun.tools.javac.tree.JCTree$JCVariableDecl.accept(JCTree.java:938)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:71)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:89)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:43)
  	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
  	at com.sun.source.util.TreeScanner.visitTry(TreeScanner.java:373)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitTry(ErrorProneScanner.java:1126)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitTry(ErrorProneScanner.java:144)
  	at com.sun.tools.javac.tree.JCTree$JCTry.accept(JCTree.java:1292)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:71)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:89)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:43)
  	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
  	at com.sun.source.util.TreeScanner.visitBlock(TreeScanner.java:248)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitBlock(ErrorProneScanner.java:575)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitBlock(ErrorProneScanner.java:144)
  	at com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:996)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:71)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:89)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:43)
  	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
  	at com.sun.source.util.TreeScanner.visitMethod(TreeScanner.java:206)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:943)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:144)
  	at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:868)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:71)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:89)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:43)
  	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
  	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
  	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
  	at com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:187)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:635)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:144)
  	at com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:778)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:71)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:89)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:43)
  	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
  	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
  	at com.sun.source.util.TreeScanner.visitCompilationUnit(TreeScanner.java:144)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:650)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:144)
  	at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:561)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:52)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:68)
  	at com.google.errorprone.scanner.ErrorProneScannerTransformer.apply(ErrorProneScannerTransformer.java:43)
  	at com.google.errorprone.ErrorProneAnalyzer.finished(ErrorProneAnalyzer.java:145)
  	at com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:120)
  	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1277)
  	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1226)
  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:840)
  	at com.sun.tools.javac.main.Main.compile(Main.java:257)
  	at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:230)
  	at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:108)
  	at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:128)
  	at com.google.errorprone.ErrorProneCompiler.compile(ErrorProneCompiler.java:70)
  	at org.codehaus.plexus.compiler.javac.errorprone.JavacCompilerWithErrorProne$CompilerInvoker.compile(JavacCompilerWithErrorProne.java:219)
  	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.lang.reflect.Method.invoke(Method.java:498)
  	at org.codehaus.plexus.compiler.javac.errorprone.JavacCompilerWithErrorProne.performCompile(JavacCompilerWithErrorProne.java:91)
  	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:943)
  	at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:137)
  	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
  	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
  	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:185)
  	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:181)
  	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  	at java.lang.Thread.run(Thread.java:745)
```

**Comment by Stephan202** [comment]
Oh man, I filed a duplicate of a ticket I created myself: #498. I need a memory upgrade.
